pkg/prom1/storage/local/chunk/chunk.go

```diff
@@ -491,6 +507,21 @@ func (it *indexAccessingChunkIterator) Value() model.SamplePair {
 	return it.lastValue
 }
 
+func (it *indexAccessingChunkIterator) Batch() Batch {
+	var batch Batch
+	j := 0
+	for j < BatchSize && it.pos < it.len {
+		batch.Timestamps[j] = int64(it.acc.timestampAtIndex(it.pos))
+		batch.Values[j] = float64(it.acc.sampleValueAtIndex(it.pos))
+		it.pos++
+		j++
+	}
+	it.pos--
```

Comment:
Is this to ensure some overlap on the next batch?

Reply:
I'll stick a comment in to explain; its because the interface contract is that you call `Scan` before calling `Batch`; without this, you'd end up skipping chunks.

Reply:
Excellent. Thanks!